### ISSUE TITLE
Image width and height in LaTeX beamer

### ISSUE THREAD

**Issue by pck1980** [jgm/pandoc#4345]
I have the following line of Pandoc markdown
`![Text](fig/some_image.png){ width=70% }\`

which Pandoc "translates" to the following LaTeX beamer code:
`\includegraphics[width=0.7\textwidth,height=\textheight]{fig/some_image.png}\\`

The problem here is that Pandoc automatically also adds the "height" option to the LaTeX code. This leads to weird (i.e. non-proportional) image scaling. If height was omitted, scaling would be fine.

The manual shows a very similar example, where the generated LaTeX code does not contain a height option.

Is there a way to stop Pandoc from adding it?

My Pandoc version is 2.1.1 (Windows)
I invoke Pandoc with
`pandoc -t beamer -o slides.tex slides.md`

**Comment by jgm** [comment]
See #4247 which also links to discussions which motivated this behavior.
Very likely, you just need to make sure that `keepaspectratio` is set as a default for `\includegraphics`  in your template, as it is in the current default template. (I assume you're using a custom template, perhaps via a `~/.pandoc/templates` directory?)

**Comment by pck1980** [comment]
Thank you for your help. As far as I can tell, I'm not using a custom template, at least not deliberately ;-)

I just tried adding the "-s" option to the command above, but this doesn't change the way "includegraphics" is invoked in the resulting tex file (i. e., "keepaspectratio" is not set).

Changing the output type from "beamer" to "latex" doesn't help either.

I searched the output of "pandoc -D beamer" for "keepaspectratio" and found a single occurrence:
`\setkeys{Gin}{width=\maxwidth,height=\maxheight,keepaspectratio}`

**Comment by jgm** [comment]
+++ Philipp [Feb 05 18 21:10 ]:
>   Thank you for your help. As far as I can tell, I'm not using a custom
>   template, at least not deliberately ;-)
>
>   I just tried adding the "-s" option to the command above, but this
>   doesn't change the way "includegraphics" is invoked in the resulting
>   tex file (i. e., "keepaspectratio" is not set).
>
>   Changing the output type from "beamer" to "latex" doesn't help either.
>
>   I searched the output of "pandoc -D beamer" for "keepaspectratio" and
>   found a single occurrence:
>   \setkeys{Gin}{width=\maxwidth,height=\maxheight,keepaspectratio}

Yes.  This sets keepaspectratio as the default, which allows
us not to specify it in the individual \includegraphics
calls.  See the linked issues for more discussion.

**Comment by pck1980** [comment]
Okay, I found my mistake. I created the TeX file without the -s option and used it as \input in some other TeX file which provides the appropriate header and footer (because this solution seemed easier than writing my own template).
This file did not set "keepaspectratio".

In addition, I misunderstood how this option is supposed to work and thought it must be specified on each call to \includegraphics, so I thought something was wrong with the output created by pandoc.

Thanks again for your help.
